addons/stock/models/stock_warehouse.py

```diff
@@ -936,11 +936,23 @@ def default_get(self, fields):
     lead_type = fields.Selection(
         [('net', 'Day(s) to get the products'), ('supplier', 'Day(s) to purchase')], 'Lead Type',
         required=True, default='supplier')
+    location_domain = fields.One2many(comodel_name='stock.location', compute='_compute_location_domain')
 
     _sql_constraints = [
         ('qty_multiple_check', 'CHECK( qty_multiple >= 0 )', 'Qty Multiple must be greater than or equal to zero.'),
     ]
 
+    @api.depends('warehouse_id')
+    def _compute_location_domain(self):
+        loc_domain = [('usage', 'in', ('internal', 'view'))]
+        # exclude other warehouses tree relation
+        other_warehouses = self.env['stock.warehouse'].search([]) - self.warehouse_id
+        for warehouse in other_warehouses:
+            loc_domain = ['&'] + loc_domain + [('parent_path', 'not like', warehouse.view_location_id.parent_path)]
```

Comment:
maybe replace with a search_read
warehouse_id != self.warehosue_id.id, parent_path

Reply:
not sure not like will work, =not like